test/ask/runtime/channel_broker_test.exs

```diff
@@ -204,14 +200,20 @@ defmodule Ask.Runtime.ChannelBrokerTest do
       # run:
       {:noreply, new_state, _} = ChannelBroker.handle_info({:collect_garbage}, state)
 
-      # it removed failed respondents (1, 3, 4) and activated queued ones (5, 6, 7):
-      assert [
-               Enum.at(respondents, 0).id,
-               Enum.at(respondents, 2).id,
-               Enum.at(respondents, 5).id,
-               Enum.at(respondents, 6).id,
-               Enum.at(respondents, 7).id
-             ] == active_respondent_ids(new_state)
+      # it removed failed respondents (1, 3, 4):
+      active_respondent_ids = active_respondent_ids(new_state)
+      assert Enum.at(respondents, 0).id in active_respondent_ids
+      assert Enum.at(respondents, 1).id not in active_respondent_ids
+      assert Enum.at(respondents, 2).id in active_respondent_ids
+      assert Enum.at(respondents, 3).id not in active_respondent_ids
+      assert Enum.at(respondents, 4).id not in active_respondent_ids
+
+      # and activated three of the pending ones (order depends on MySQL version):
+      assert_some_in_both(
+        3,
+        respondents |> Enum.drop(5) |> Enum.map(& &1.id),
+        active_respondent_ids
+      )
```

Comment:
Alternatively, maybe MySQL 8 finally allows to have nanosecond precision? That could fix the ordering issue if `queue_at` had nanosecond precision (but only on that field, as it would likely break things in other places in Surveda) :thinking:

Reply:
> MySQL has fractional seconds support for `TIME`, `DATETIME`, and `TIMESTAMP` values, with up to microseconds (6 digits) precision
[13.2.6 Fractional Seconds in Time Values](https://dev.mysql.com/doc/refman/8.0/en/fractional-seconds.html)

So no microseconds yet.

But I like the expressiveness of your solution. I'll do that 👍